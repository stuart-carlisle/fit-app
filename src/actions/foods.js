import { database } from '../firebase/firebase'

//SET_FOODS_PUBLIC
export const setFoodsPublic = (foodsPublic) => ({
    type:'SET_FOODS_PUBLIC',
    foodsPublic
  })


//SET_FOODS_PRIVATE
export const setFoodsPrivate = (foodsPrivate) => ({
  type:'SET_FOODS_PRIVATE',
  foodsPrivate
})

export const startSetFoodsPrivate = () => {
  return (dispatch,getState) => {
    const uid = getState().auth.uid
    return database.ref(`users/${uid}/foods`).once('value').then((snapshot)=>{
        const foodsPrivate = []
        snapshot.forEach((childSnapshot)=>{
          foodsPrivate.push({
            id:childSnapshot.key,
            ...childSnapshot.val()
          })
        })
        dispatch(setFoodsPrivate(foodsPrivate))
    })  
  }
}

//SET_FOODS_DIARY
export const setFoodsDiary = (foods) => ({
  type:'SET_FOODS_DIARY',
  foods
})

export const startSetFoodsDiary = () => {
  return (dispatch,getState) => {
    const uid = getState().auth.uid
    return database.ref(`users/${uid}/diary/foods`).once('value').then((snapshot)=>{
        const foods = []
        snapshot.forEach((childSnapshot)=>{
          foods.push({
            id:childSnapshot.key,
            ...childSnapshot.val()
          })
        })
        dispatch(setFoodsDiary(foods))
    })  
  }
}

// ADD_FOOD PRIVATE
export const addFoodPrivate = (food) => ({
  type: 'ADD_FOOD_PRIVATE',
  food
});

export const startAddFoodPrivate = (foodData) => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid
    const { description, serving, nutrition, lastUsed } = foodData
    const food = { description, serving, nutrition, lastUsed }
   
    return database.ref(`users/${uid}/foods`).push(food).then((ref)=>{
      dispatch(addFoodPrivate({
        id:ref.key,
        ...food
      }))
    })
  }
}

// ADD_FOOD_DIARY
export const addFoodDiary = (food) => ({
  type: 'ADD_FOOD_DIARY',
  food
});

export const startAddFoodDiary = (foodData) => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid
    const {description, serving, nutrition, diaryDate, numberOfServings, meal} = foodData
    const food = { description, serving, nutrition, diaryDate, numberOfServings, meal}
   
    return database.ref(`users/${uid}/diary/foods`).push(food).then((ref)=>{
      dispatch(addFoodDiary({
        id: ref.key,
        ...food
      }))
    })
  }
}

// UPDATE_FOOD_PRIVATE
export const updateFoodPrivate = (id, updates) => ({
  type: 'UPDATE_FOOD_PRIVATE',
  id,
  updates
});

export const startUpdateFoodPrivate = (id, updates) => {
  return (dispatch,getState) => {
    const uid = getState().auth.uid
    return database.ref(`users/${uid}/foods/${id}`).update(updates).then(()=>{
      dispatch(updateFoodPrivate(id,updates))
    })
  }
}

// REMOVE_FOOD_DIARY
export const removeFoodDiary = ({ id } = {}) => ({
  type: 'REMOVE_FOOD_DIARY',
  id
});


export const startRemoveFoodDiary = ({ id } = {}) => {
  return (dispatch,getState) => {
    const uid = getState().auth.uid
    return database.ref(`users/${uid}/diary/foods/${id}`).remove().then(()=>{
      dispatch(removeFoodDiary({id}))
    })
  }
}

// UPDATE_FOOD_DIARY
export const updateFoodDiary = (id, updates) => ({
  type: 'UPDATE_FOOD_DIARY',
  id,
  updates
});

export const startUpdateFoodDiary = (id, updates) => {
  return (dispatch,getState) => {
  const uid = getState().auth.uid
  return database.ref(`users/${uid}/diary/foods/${id}`).update(updates).then(()=>{
    dispatch(updateFoodDiary(id,updates))
  })
}}

// UPDATE_TARGETS
export const updateTargets = (updates) => ({
  type: 'UPDATE_TARGETS',
  updates
})

export const startUpdateTargets = (updates) => {
  return (dispatch,getState) => {
  const uid = getState().auth.uid
  return database.ref(`users/${uid}/diary/targets`).update(updates).then(()=>{
    dispatch(updateTargets(updates))
  })
}}

// SET_TARGETS
export const setTargets = (targets) => ({
  type:'SET_TARGETS',
  targets
})

export const startSetTargets = () => {
  return (dispatch,getState) => {
    const uid = getState().auth.uid
    return database.ref(`users/${uid}/diary/targets`).once('value').then((snapshot)=>{
      const targets = snapshot.val()
      if(!!targets){
      dispatch(setTargets(targets))
      }
  })  
}
}

// UPDATE_WEIGHT
export const updateWeight = (weight) => ({
  type: 'UPDATE_WEIGHT',
  weight
})

export const startUpdateWeight = (weight,date) => {
  return (dispatch,getState) => {
  const uid = getState().auth.uid
  return database.ref(`users/${uid}/diary/weights/${date}`).set({weight})
  .then(()=>{
    dispatch(updateWeight({weight: weight, date: date}))
  })
}}

//SET_WEIGHTS_DIARY
export const setWeightsDiary = (weights) => ({
  type:'SET_WEIGHTS_DIARY',
  weights
})

export const startSetWeightsDiary = () => {
  return (dispatch,getState) => {
    const uid = getState().auth.uid
    return database.ref(`users/${uid}/diary/weights`).once('value').then((snapshot)=>{
        const weights = []
        snapshot.forEach((childSnapshot)=>{
          weights.push({
            ...childSnapshot.val(),
            date:childSnapshot.key
          })
        })
        dispatch(setWeightsDiary(weights))
    })  
  }
}