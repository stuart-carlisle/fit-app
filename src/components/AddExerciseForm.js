import React, { useState , useEffect} from 'react'
import moment from 'moment'
import { connect } from 'react-redux'
import numeral from 'numeral'

export const AddExerciseForm = ({
  filters = {},
  onSubmit, 
  exercise = {
    description:'',
    type:'',
    energy:'',
    time:''
  }
}) => {
    const defaultEnergy = numeral(exercise.energy).value()
    const defaultTime = numeral(exercise.time).value()
    const date = filters.date
    const referenceDatabase = filters.referenceDatabase
    const [time, setTime] = useState(defaultTime)
    const [error, setError] = useState('')
    const [energy, setEnergy] = useState(defaultEnergy)
    const diaryDate = date ? date.valueOf() : moment().startOf('day').valueOf()
    const lastUsed = moment().valueOf()

    useEffect(()=>{
        
      let refreshed = JSON.parse(localStorage.getItem('refreshed'))
      if(refreshed){
        window.history.back()
        window.scrollTo(0,0)
        localStorage.removeItem('refreshed')
      }else{
        localStorage.setItem('refreshed',JSON.stringify(true))
      }
    },[])

    const onFormSubmit = (e) => {
        e.preventDefault();
        const exerciseData = { 
          description: exercise.description,
          type: exercise.type
        }
        if (numeral(time).value()<=0) {
          setError('Please provide a time greater than 0')
          window.scrollTo(0,0)
        } else {
          setError('')
          onSubmit(
            referenceDatabase,
            diaryDate,
            lastUsed,
            exerciseData,
            time.toString(),
            energy.toString()
          )
          localStorage.removeItem('refreshed')
        }
      }
    
    const onTimeChange = (e)=>{
        const time = e.target.value
          if (!time || time.match(/^\d{1,}(\.\d{0,1})?$/)) {
          setTime(time)
          const timeValue = numeral(time).value()
          setEnergy(Math.round(defaultEnergy * (timeValue/defaultTime)))
        }
    }

    const onEnergyChange = ()=>{
      
    }

    return (
        <div>
        <div className="content-container__background">
        <div className="content-container__form__update">
            <form className= "form" onSubmit={onFormSubmit}>
                {error && <p id="error-message" className="form__error">{error}</p>}
                  <h1 className="page-header__title-update"> {exercise.description} </h1>
                    <span className="text-input__span-container-form">
                      <div className="content-container__update-titles">
                        <h3 className="form__heading__update">Exercise Time (mins):</h3>
                        <h3 className="form__heading__update">Energy (calories): </h3>
                      </div>
                      <div className="content-container__update">
                        <input  //Exercise Duration
                          className="text-input__update-form"
                          type="text"
                          value={time}
                          onChange={onTimeChange}
                          placeholder="Enter a Duration"
                        />  
                        <input  //Energy Used
                          className="text-input__update-form"
                          type="text"
                          value={energy?energy:''}
                          onChange={onEnergyChange}
                          placeholder="Energy Used"
                        /> 
                      </div>
                    </span>
                <button className="button button--form">Add Exercise</button>
            </form>
          </div>
        </div>
      </div>  
    )
}

const mapStateToProps = (state) => {
  return {
    filters: state.filters
  }
}


export default connect(mapStateToProps)(AddExerciseForm);