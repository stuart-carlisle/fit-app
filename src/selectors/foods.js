// Get visible expenses

export default (foods, { text, sortBy }) => {

  return foods.filter((food) => {
    const textMatch = food.description.toLowerCase().includes(text.toLowerCase())
    return textMatch;
  }).sort((a, b) => {
    if (sortBy === 'date') {
      return a.lastUsed < b.lastUsed ? 1 : -1;
    } else if (sortBy === 'alphabetical') {
      return a.description > b.description ? 1 : -1;
    }
  })
} 
