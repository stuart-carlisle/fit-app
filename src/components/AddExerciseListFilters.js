import React, { useState } from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByDate, sortByAlphabetical, setReferenceDatabaseToPublic } from '../actions/filters';
import { setExercisesPublic } from '../actions/exercises'
import algoliasearch from 'algoliasearch/lite'

export const AddExerciseListFilters = ({
    filters,
    setTextFilter, 
    setReferenceDatabasePublic,
    sortByDate, 
    sortByAlphabetical,
    setExercisesPublic 
  }) => {
 
  const searchClient = algoliasearch (
      process.env.ALGOLIA_APP_ID,
      process.env.ALGOLIA_API_KEY
  )

  const index = searchClient.initIndex('exercises');

  const [text, setText] = useState('')
    
  const onTextChange = (e) => {
    setTextFilter(e.target.value)
    setText(e.target.value)
  }

  const onSortChange = (e) => {
    if (e.target.value === 'date') {
      sortByDate()
    } else if (e.target.value === 'alphabetical') {
      sortByAlphabetical()
    }
  }

  const onSubmit = (e) => {
    e.preventDefault()
    index
      .search({
        query: text
      })
      .then((responses) => {
        const array = responses.hits
        const newExercises = array.map((exercise)=>{
          return{
            description:exercise.description,
            type:exercise.type,
            energy:exercise.energy,
            time:exercise.time,
            id:exercise.objectID
          }
        })
        setExercisesPublic(newExercises)
      }).then(()=>{
        setReferenceDatabasePublic()
      })
}

    return (
      
      <div className="header__content--filters"> 
      <form className="filters-text-input__span-container" onSubmit={onSubmit}>
          <div className="input-group__item-page-header">
            <input
              type="text"
              className="filters-text-input"
              placeholder="SEARCH EXERCISES"
              value={text}
              onChange={onTextChange}
            />
          </div>
          <div className="input-group__item-page-header">
            <select
              className="minimal"
              value={filters.sortBy}
              onChange={onSortChange}
            >
              <option value="date">DATE</option>
              <option value="alphabetical">ALPHABETICAL</option>
            </select>
          </div>
            <button className="button filters-button--form-page-header">Search Online</button>
            </form>   
        </div>
    )
}

const mapStateToProps = (state) => {
  return {
    filters: state.filters
  };
};

const mapDispatchToProps = (dispatch) => ({
  setTextFilter: (text) => dispatch(setTextFilter(text)),
  sortByDate: () => dispatch(sortByDate()),
  sortByAlphabetical: () => dispatch(sortByAlphabetical()),
  setReferenceDatabasePublic: () => dispatch(setReferenceDatabaseToPublic()),
  setExercisesPublic: (exercises) => dispatch(setExercisesPublic(exercises))
})

export default connect(mapStateToProps,mapDispatchToProps)(AddExerciseListFilters);

