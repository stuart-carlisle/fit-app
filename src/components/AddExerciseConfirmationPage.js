import React from 'react'
import { connect } from 'react-redux'
import AddExerciseForm from './AddExerciseForm'
import { Link } from 'react-router-dom'
import { startAddExerciseDiary, startAddExercisePrivate, startUpdateExercisePrivate } from '../actions/exercises'

export const AddExerciseConfirmationPage = (props) => {
  
  const onSubmit = (referenceDatabase, diaryDate, lastUsed, exerciseDescAndType, time, energy) => {
    if (referenceDatabase==='public'){
      props.startAddExercisePrivate({...exerciseDescAndType, time, energy, lastUsed})
    } else if (referenceDatabase==='private'){
      props.startUpdateExercisePrivate(props.exercise.id,{...exerciseDescAndType, time, energy, lastUsed})
    }
    props.startAddExerciseDiary({...exerciseDescAndType, time, diaryDate, energy})
    props.history.push('/diary');
  }
  

  return (
     <div>
     <Link className="button__back-button" to={`/add-exercise`}><img className="image__header" src="/images/back-arrow.svg"/></Link>
        <div className="page-header__filter page-header__filter--with-back">
          <div className="content-container">
            <h1 className="page-header__title">ADD EXERCISE</h1>
          </div>
        </div>
        <div className="show-for-mobile page-header__margin"></div>
        <div className="content-container">
        <AddExerciseForm id="add-exercise-form" exercise={props.exercise} onSubmit={onSubmit} />
        </div>
     </div>
 )
}

const mapStateToProps = (state,props) => {
  if(state.filters.referenceDatabase==='public'){
    return {
      exercise: state.exercisesPublic.find((exercise) => exercise.id === props.match.params.id)
    }
  }else if(state.filters.referenceDatabase==='private'){
    return {
      exercise: state.exercisesPrivate.find((exercise) => exercise.id === props.match.params.id)
    }
  }
}

const mapDispatchToProps = (dispatch) =>({
    startAddExerciseDiary: (exercise) => dispatch(startAddExerciseDiary(exercise)),
    startUpdateExercisePrivate: (id, exercise) => dispatch(startUpdateExercisePrivate(id, exercise)),
    startAddExercisePrivate: (exercise) => dispatch(startAddExercisePrivate(exercise))
})

export default connect(mapStateToProps, mapDispatchToProps)(AddExerciseConfirmationPage);




