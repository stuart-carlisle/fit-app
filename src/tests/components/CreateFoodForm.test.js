import React from 'react'
import { shallow } from 'enzyme'
import { CreateFoodForm  } from '../../components/CreateFoodForm'
import moment from 'moment'
import foods, {altFoods} from '../fixtures/foods'


test('Should render food form correctly',()=>{
    const wrapper = shallow(<CreateFoodForm />)
    expect(wrapper).toMatchSnapshot()
})

test('Should render food form with filter data',()=>{
    const filters={
        date:moment(0).startOf('day'),
        text: '',
        sortBy: 'date',
        referenceDatabase: 'private',
        chartType: {
            chart: 'calorie-tracker', 
            option: 'week'
        },
        meal: 'breakfast'
    }
    const wrapper = shallow(<CreateFoodForm filters={filters}/>)
    expect(wrapper).toMatchSnapshot()
})

test('Should render error for invalid form submission',()=>{
    const wrapper = shallow(<CreateFoodForm />)
    wrapper.find('form').simulate('submit',{
        preventDefault:() => { }
    })
    expect(wrapper.exists('#error-message')).toBe(true);
})

test('Should render correct value when description is changed',()=>{
    const wrapper = shallow(<CreateFoodForm />)
    const value = 'bread'
    wrapper.find('input').at(0).simulate('change',{
        target: { value }
    })
    expect(wrapper.find('input').at(0).prop('value')).toBe(value)
})

test('Should render correct value when serving size is changed',()=>{
    const wrapper = shallow(<CreateFoodForm />)
    const servingSize = '14.5'
    const unit = 'oz'
    wrapper.find('input').at(1).simulate('change',{
        target: { value: servingSize }
    })
    wrapper.find('select').at(0).simulate('change',{
        target: { value: unit }
    })
    expect(wrapper.find('input').at(1).prop('value')).toBe(servingSize)
    expect(wrapper.find('select').at(0).prop('value')).toBe(unit)
})

test('Should set serving size given invalid data',()=>{
    const wrapper = shallow(<CreateFoodForm />)
    const servingSize = '14.5999'
    const unit = 'g'
    wrapper.find('input').at(1).simulate('change',{
        target: { value: servingSize }
    })
    wrapper.find('select').at(0).simulate('change',{
        target: { value: unit }
    })
    expect(wrapper.find('input').at(1).prop('value')).toBe('')
    expect(wrapper.find('select').at(0).prop('value')).toBe(unit)
})

test('Should render correct value when servingsPerContainer is changed',()=>{
    const wrapper = shallow(<CreateFoodForm />)
    const value = '2'
    wrapper.find('input').at(2).simulate('change',{
        target: { value }
    })
    expect(wrapper.find('input').at(2).prop('value')).toBe(value)
})

test('Should set servings per container given invalid data',()=>{
    const wrapper = shallow(<CreateFoodForm />)
    const value = '2.459'
    wrapper.find('input').at(2).simulate('change',{
        target: { value }
    })
    expect(wrapper.find('input').at(2).prop('value')).toBe('')
})

test('Should render correct value when energy is changed',()=>{
    const wrapper = shallow(<CreateFoodForm />)
    const value = '289'
    wrapper.find('input').at(3).simulate('change',{
        target: { value }
    })
    expect(wrapper.find('input').at(3).prop('value')).toBe(value)
})

test('Should set energy given invalid data',()=>{
    const wrapper = shallow(<CreateFoodForm />)
    const value = '276.8878'
    wrapper.find('input').at(3).simulate('change',{
        target: { value }
    })
    expect(wrapper.find('input').at(3).prop('value')).toBe('')
})

test('Should render correct value when total carbs is changed',()=>{
    const wrapper = shallow(<CreateFoodForm />)
    const totalCarbs = '14.5'
    const unit = 'g'
    wrapper.find('input').at(4).simulate('change',{
        target: { value: totalCarbs }
    })
    wrapper.find('select').at(1).simulate('change',{
        target: { value: unit }
    })
    expect(wrapper.find('input').at(4).prop('value')).toBe(totalCarbs)
    expect(wrapper.find('select').at(1).prop('value')).toBe(unit)
})

test('Should set total carbs given invalid data',()=>{
    const wrapper = shallow(<CreateFoodForm />)
    const totalCarbs = '14.5999'
    const unit = 'g'
    wrapper.find('input').at(4).simulate('change',{
        target: { value: totalCarbs }
    })
    wrapper.find('select').at(1).simulate('change',{
        target: { value: unit }
    })
    expect(wrapper.find('input').at(4).prop('value')).toBe('')
    expect(wrapper.find('select').at(1).prop('value')).toBe(unit)
})

test('Should render correct value when sugars is changed',()=>{
    const wrapper = shallow(<CreateFoodForm />)
    const sugars = '14.5'
    const unit = 'g'
    wrapper.find('input').at(5).simulate('change',{
        target: { value: sugars }
    })
    wrapper.find('select').at(2).simulate('change',{
        target: { value: unit }
    })
    expect(wrapper.find('input').at(5).prop('value')).toBe(sugars)
    expect(wrapper.find('select').at(2).prop('value')).toBe(unit)
})

test('Should set sugars given invalid data',()=>{
    const wrapper = shallow(<CreateFoodForm />)
    const sugars = '14.5999'
    const unit = 'g'
    wrapper.find('input').at(5).simulate('change',{
        target: { value: sugars }
    })
    wrapper.find('select').at(2).simulate('change',{
        target: { value: unit }
    })
    expect(wrapper.find('input').at(5).prop('value')).toBe('')
    expect(wrapper.find('select').at(2).prop('value')).toBe(unit)
})

test('Should render correct value when protein is changed',()=>{
    const wrapper = shallow(<CreateFoodForm />)
    const protein = '14.5'
    const unit = 'g'
    wrapper.find('input').at(6).simulate('change',{
        target: { value: protein }
    })
    wrapper.find('select').at(3).simulate('change',{
        target: { value: unit }
    })
    expect(wrapper.find('input').at(6).prop('value')).toBe(protein)
    expect(wrapper.find('select').at(3).prop('value')).toBe(unit)
})

test('Should set sugars given invalid data',()=>{
    const wrapper = shallow(<CreateFoodForm />)
    const protein = '14.5999'
    const unit = 'g'
    wrapper.find('input').at(6).simulate('change',{
        target: { value: protein }
    })
    wrapper.find('select').at(3).simulate('change',{
        target: { value: unit }
    })
    expect(wrapper.find('input').at(6).prop('value')).toBe('')
    expect(wrapper.find('select').at(3).prop('value')).toBe(unit)
})

test('Should render correct value when total fat is changed',()=>{
    const wrapper = shallow(<CreateFoodForm />)
    const totalFat = '14.5'
    const unit = 'g'
    wrapper.find('input').at(7).simulate('change',{
        target: { value: totalFat }
    })
    wrapper.find('select').at(4).simulate('change',{
        target: { value: unit }
    })
    expect(wrapper.find('input').at(7).prop('value')).toBe(totalFat)
    expect(wrapper.find('select').at(4).prop('value')).toBe(unit)
})

test('Should set totalFat given invalid data',()=>{
    const wrapper = shallow(<CreateFoodForm />)
    const totalFat = '14.5999'
    const unit = 'g'
    wrapper.find('input').at(7).simulate('change',{
        target: { value: totalFat }
    })
    wrapper.find('select').at(4).simulate('change',{
        target: { value: unit }
    })
    expect(wrapper.find('input').at(7).prop('value')).toBe('')
    expect(wrapper.find('select').at(4).prop('value')).toBe(unit)
})

test('Should render correct value when sat fat is changed',()=>{
    const wrapper = shallow(<CreateFoodForm />)
    const satFat = '14.5'
    const unit = 'g'
    wrapper.find('input').at(8).simulate('change',{
        target: { value: satFat }
    })
    wrapper.find('select').at(5).simulate('change',{
        target: { value: unit }
    })
    expect(wrapper.find('input').at(8).prop('value')).toBe(satFat)
    expect(wrapper.find('select').at(5).prop('value')).toBe(unit)
})

test('Should set satFat given invalid data',()=>{
    const wrapper = shallow(<CreateFoodForm />)
    const satFat = '14.5999'
    const unit = 'g'
    wrapper.find('input').at(8).simulate('change',{
        target: { value: satFat }
    })
    wrapper.find('select').at(5).simulate('change',{
        target: { value: unit }
    })
    expect(wrapper.find('input').at(8).prop('value')).toBe('')
    expect(wrapper.find('select').at(5).prop('value')).toBe(unit)
})

test('Should simulate form completion and call onSubmit prop for valid form submission',()=>{
    const filters={
        date:moment(0).startOf('day'),
        text: '',
        sortBy: 'date',
        referenceDatabase: 'private',
        chartType: {
            chart: 'calorie-tracker', 
            option: 'week'
        },
        meal: 'breakfast'
    }
    const onSubmitSpy = jest.fn()
    const wrapper = shallow(<CreateFoodForm filters={filters} onSubmit={onSubmitSpy}/>)
    
    wrapper.find('input').at(0).simulate('change',{//description
        target: { value: foods[0].description }
    })
    wrapper.find('input').at(1).simulate('change',{//serving size
        target: { value: foods[0].serving.servingSize.size }
    })
    wrapper.find('select').at(0).simulate('change',{//serving size unit
        target: { value: foods[0].serving.servingSize.unit }
    })
    wrapper.find('input').at(3).simulate('change',{//energy
        target: { value: foods[0].nutrition.energy.amount }
    })
    
    wrapper.find('form').simulate('submit',{
        preventDefault:() => { }
    })
    expect(wrapper.exists('#error-message')).toBe(false);
    expect(onSubmitSpy).toHaveBeenLastCalledWith({
        description:foods[0].description,
        serving:foods[0].serving,
        nutrition:foods[0].nutrition,
        diaryDate:filters.date.valueOf(),
        lastUsed:moment().valueOf(),
        numberOfServings: '1',
        meal: filters.meal
    })
})

test('Should simulate form submission converting oz to g correctly',()=>{
    const filters={
        date:moment(0).startOf('day'),
        text: '',
        sortBy: 'date',
        referenceDatabase: 'private',
        chartType: {
            chart: 'calorie-tracker', 
            option: 'week'
        },
        meal: 'breakfast'
    }
    const onSubmitSpy = jest.fn()
    const wrapper = shallow(<CreateFoodForm filters={filters} onSubmit={onSubmitSpy}/>)
    
    wrapper.find('input').at(0).simulate('change',{//description
        target: { value: altFoods[0].description }
    })
    wrapper.find('input').at(1).simulate('change',{//serving size
        target: { value: altFoods[0].serving.servingSize.size }
    })
    wrapper.find('select').at(0).simulate('change',{//serving size unit
        target: { value: altFoods[0].serving.servingSize.unit }
    })
    wrapper.find('input').at(2).simulate('change',{//servingsPerContainer
        target: { value: altFoods[0].serving.servingsPerContainer }
    })
    wrapper.find('input').at(3).simulate('change',{//energy
        target: { value: altFoods[0].nutrition.energy.amount }
    })
    wrapper.find('input').at(4).simulate('change',{//totalCarbs
        target: { value: altFoods[0].nutrition.totalCarbs.amount }
    })
    wrapper.find('select').at(1).simulate('change',{//totalCarbs unit
        target: { value: altFoods[0].nutrition.totalCarbs.unit }
    })
    wrapper.find('input').at(5).simulate('change',{//sugars
        target: { value: altFoods[0].nutrition.sugars.amount }
    })
    wrapper.find('select').at(2).simulate('change',{//sugars unit
        target: { value: altFoods[0].nutrition.sugars.unit }
    })
    wrapper.find('input').at(6).simulate('change',{//protein
        target: { value: altFoods[0].nutrition.protein.amount }
    })
    wrapper.find('select').at(3).simulate('change',{//protein unit
        target: { value: altFoods[0].nutrition.protein.unit }
    })
    wrapper.find('input').at(7).simulate('change',{//totalFat
        target: { value: altFoods[0].nutrition.totalFat.amount }
    })
    wrapper.find('select').at(4).simulate('change',{//totalFat unit
        target: { value: altFoods[0].nutrition.totalFat.unit }
    })
    wrapper.find('input').at(8).simulate('change',{//satFat
        target: { value: altFoods[0].nutrition.satFat.amount }
    })
    wrapper.find('select').at(5).simulate('change',{//satFat unit
        target: { value: altFoods[0].nutrition.satFat.unit }
    })
    
    wrapper.find('form').simulate('submit',{
        preventDefault:() => { }
    })
    expect(wrapper.exists('#error-message')).toBe(false);
    expect(onSubmitSpy).toHaveBeenLastCalledWith({
        description:altFoods[1].description,
        serving:altFoods[1].serving,
        nutrition:altFoods[1].nutrition,
        diaryDate:filters.date.valueOf(),
        lastUsed:moment().valueOf(),
        numberOfServings: '1',
        meal: filters.meal
    })
})