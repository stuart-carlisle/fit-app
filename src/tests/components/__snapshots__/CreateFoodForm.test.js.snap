// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render food form correctly 1`] = `
<div>
  <div
    className="content-container__background"
  >
    <div
      className="content-container__form"
    >
      <form
        className="form"
        onSubmit={[Function]}
      >
        <input
          className="text-input"
          onChange={[Function]}
          placeholder="Description"
          type="text"
          value=""
        />
        <h3
          className="form__heading"
        >
          SERVING INFORMATION
        </h3>
        <span
          className="text-input__span-container-form"
        >
          <input
            className="text-input"
            onChange={[Function]}
            placeholder="Serving Size"
            type="text"
            value=""
          />
          <select
            className="form-input-select"
            onChange={[Function]}
            required={false}
            value=""
          >
            <option
              hidden={true}
              value=""
            >
              Unit
            </option>
            <option
              value="g"
            >
              grams
            </option>
            <option
              value="oz"
            >
              ounces
            </option>
          </select>
        </span>
        <input
          className="text-input"
          onChange={[Function]}
          placeholder="Servings Per Container"
          type="text"
          value=""
        />
        <h3
          className="form__heading"
        >
          NUTRIONAL INFORMATION
        </h3>
        <input
          className="text-input"
          onChange={[Function]}
          placeholder="Energy in Kilo Calories"
          type="text"
          value=""
        />
        <span
          className="text-input__span-container-form"
        >
          <input
            className="text-input"
            onChange={[Function]}
            placeholder="Total Carbohydrates"
            type="text"
            value=""
          />
          <select
            className="form-input-select"
            onChange={[Function]}
            required={false}
            value=""
          >
            <option
              hidden={true}
              value=""
            >
              Unit
            </option>
            <option
              value="g"
            >
              grams
            </option>
            <option
              value="oz"
            >
              ounces
            </option>
          </select>
        </span>
        <span
          className="text-input__span-container-form"
        >
          <input
            className="text-input"
            onChange={[Function]}
            placeholder="Sugars"
            type="text"
            value=""
          />
          <select
            className="form-input-select"
            onChange={[Function]}
            required={false}
            value=""
          >
            <option
              hidden={true}
              value=""
            >
              Unit
            </option>
            <option
              value="g"
            >
              grams
            </option>
            <option
              value="oz"
            >
              ounces
            </option>
          </select>
        </span>
        <span
          className="text-input__span-container-form"
        >
          <input
            className="text-input"
            onChange={[Function]}
            placeholder="Protein"
            type="text"
            value=""
          />
          <select
            className="form-input-select"
            onChange={[Function]}
            required={false}
            value=""
          >
            <option
              hidden={true}
              value=""
            >
              Unit
            </option>
            <option
              value="g"
            >
              grams
            </option>
            <option
              value="oz"
            >
              ounces
            </option>
          </select>
        </span>
        <span
          className="text-input__span-container-form"
        >
          <input
            className="text-input"
            onChange={[Function]}
            placeholder="Total Fat"
            type="text"
            value=""
          />
          <select
            className="form-input-select"
            onChange={[Function]}
            required={false}
            value=""
          >
            <option
              hidden={true}
              value=""
            >
              Unit
            </option>
            <option
              value="g"
            >
              grams
            </option>
            <option
              value="oz"
            >
              ounces
            </option>
          </select>
        </span>
        <span
          className="text-input__span-container-form"
        >
          <input
            className="text-input"
            onChange={[Function]}
            placeholder="Saturated Fat"
            type="text"
            value=""
          />
          <select
            className="form-input-select"
            onChange={[Function]}
            required={false}
            value=""
          >
            <option
              hidden={true}
              value=""
            >
              Unit
            </option>
            <option
              value="g"
            >
              grams
            </option>
            <option
              value="oz"
            >
              ounces
            </option>
          </select>
        </span>
        <button
          className="button button--form"
        >
          Add Food
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`Should render food form with filter data 1`] = `
<div>
  <div
    className="content-container__background"
  >
    <div
      className="content-container__form"
    >
      <form
        className="form"
        onSubmit={[Function]}
      >
        <input
          className="text-input"
          onChange={[Function]}
          placeholder="Description"
          type="text"
          value=""
        />
        <h3
          className="form__heading"
        >
          SERVING INFORMATION
        </h3>
        <span
          className="text-input__span-container-form"
        >
          <input
            className="text-input"
            onChange={[Function]}
            placeholder="Serving Size"
            type="text"
            value=""
          />
          <select
            className="form-input-select"
            onChange={[Function]}
            required={false}
            value=""
          >
            <option
              hidden={true}
              value=""
            >
              Unit
            </option>
            <option
              value="g"
            >
              grams
            </option>
            <option
              value="oz"
            >
              ounces
            </option>
          </select>
        </span>
        <input
          className="text-input"
          onChange={[Function]}
          placeholder="Servings Per Container"
          type="text"
          value=""
        />
        <h3
          className="form__heading"
        >
          NUTRIONAL INFORMATION
        </h3>
        <input
          className="text-input"
          onChange={[Function]}
          placeholder="Energy in Kilo Calories"
          type="text"
          value=""
        />
        <span
          className="text-input__span-container-form"
        >
          <input
            className="text-input"
            onChange={[Function]}
            placeholder="Total Carbohydrates"
            type="text"
            value=""
          />
          <select
            className="form-input-select"
            onChange={[Function]}
            required={false}
            value=""
          >
            <option
              hidden={true}
              value=""
            >
              Unit
            </option>
            <option
              value="g"
            >
              grams
            </option>
            <option
              value="oz"
            >
              ounces
            </option>
          </select>
        </span>
        <span
          className="text-input__span-container-form"
        >
          <input
            className="text-input"
            onChange={[Function]}
            placeholder="Sugars"
            type="text"
            value=""
          />
          <select
            className="form-input-select"
            onChange={[Function]}
            required={false}
            value=""
          >
            <option
              hidden={true}
              value=""
            >
              Unit
            </option>
            <option
              value="g"
            >
              grams
            </option>
            <option
              value="oz"
            >
              ounces
            </option>
          </select>
        </span>
        <span
          className="text-input__span-container-form"
        >
          <input
            className="text-input"
            onChange={[Function]}
            placeholder="Protein"
            type="text"
            value=""
          />
          <select
            className="form-input-select"
            onChange={[Function]}
            required={false}
            value=""
          >
            <option
              hidden={true}
              value=""
            >
              Unit
            </option>
            <option
              value="g"
            >
              grams
            </option>
            <option
              value="oz"
            >
              ounces
            </option>
          </select>
        </span>
        <span
          className="text-input__span-container-form"
        >
          <input
            className="text-input"
            onChange={[Function]}
            placeholder="Total Fat"
            type="text"
            value=""
          />
          <select
            className="form-input-select"
            onChange={[Function]}
            required={false}
            value=""
          >
            <option
              hidden={true}
              value=""
            >
              Unit
            </option>
            <option
              value="g"
            >
              grams
            </option>
            <option
              value="oz"
            >
              ounces
            </option>
          </select>
        </span>
        <span
          className="text-input__span-container-form"
        >
          <input
            className="text-input"
            onChange={[Function]}
            placeholder="Saturated Fat"
            type="text"
            value=""
          />
          <select
            className="form-input-select"
            onChange={[Function]}
            required={false}
            value=""
          >
            <option
              hidden={true}
              value=""
            >
              Unit
            </option>
            <option
              value="g"
            >
              grams
            </option>
            <option
              value="oz"
            >
              ounces
            </option>
          </select>
        </span>
        <button
          className="button button--form"
        >
          Add Food
        </button>
      </form>
    </div>
  </div>
</div>
`;
